sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOIT.java

```diff
@@ -0,0 +1,190 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.jdbc;
+
+import static org.junit.Assert.assertEquals;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import javax.sql.DataSource;
+
+import org.apache.beam.sdk.coders.BigEndianIntegerCoder;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.postgresql.ds.PGSimpleDataSource;
+
+
+/**
+ * A test of JdbcIO on an independent Postgres instance.
+ *
+ * <p>This test requires a running instance of Postgres, and the test dataset must exist in the
+ * databaase. `JdbcTestDataSet` will create the read table.
+ *
+ * <p>You can run just this test by doing the following:
+ * mvn test-compile compile failsafe:integration-test -D beamTestPipelineOptions='[
+ * "--postgresServerName=1.2.3.4",
+ * "--postgresUsername=postgres",
+ * "--postgresDatabaseName=myfancydb",
+ * "--postgresPassword=yourpassword",
+ * "--postgresSsl=false"
+ * ]' -DskipITs=false -Dit.test=org.apache.beam.sdk.io.jdbc.JdbcIOIT -DfailIfNoTests=false
+ */
+@RunWith(JUnit4.class)
+public class JdbcIOIT {
+  public PGSimpleDataSource getDataSource() throws SQLException {
+    PipelineOptionsFactory.register(PostgresTestOptions.class);
+    PostgresTestOptions options = TestPipeline.testingPipelineOptions()
+        .as(PostgresTestOptions.class);
+
+    return JdbcTestDataSet.getDataSource(options);
+  }
+  private static class CreateKVOfNameAndId implements JdbcIO.RowMapper<KV<String, Integer>> {
+    @Override
+    public KV<String, Integer> mapRow(ResultSet resultSet) throws Exception {
+      KV<String, Integer> kv =
+          KV.of(resultSet.getString("name"), resultSet.getInt("id"));
+      return kv;
+    }
+  }
+
+  private static class AssertCountFn implements
+      SerializableFunction<Iterable<KV<String, Long>>, Void> {
+    public AssertCountFn(long expectedCount) {
+     this.expectedCount = expectedCount;
+    }
+
+    private long expectedCount;
+
+    @Override
+    public Void apply(Iterable<KV<String, Long>> input) {
+      for (KV<String, Long> element : input) {
+        assertEquals(
+            element.getKey(), expectedCount, element.getValue().longValue());
+      }
+      return null;
+    }
+  }
```

Comment:
is there a reason to do `PAssert.satisfies` instead of `PAssert.containsInAnyOrder`? It looks to me like the latter would simplify much of this logic.